cmake/partition_manager.cmake

```diff
@@ -73,6 +73,25 @@ if (NOT static_configuration AND CONFIG_PM_IMAGE_NOT_BUILT_FROM_SOURCE)
     place the image correctly in flash.")
 endif()
 
+# From Documentation: By default, the Partition Manager dynamically
+# places the partitions in memory. However, if you have a deployed
+# product that consists of multiple images, where only a subset of
+# the included images can be upgraded through a firmware update
+# mechanism, the upgradable images must be statically configured.
+if (NOT static_configuration AND (
+  (CONFIG_BOOTLOADER_MCUBOOT  AND NOT CONFIG_SECURE_BOOT        ) OR
+  (CONFIG_SECURE_BOOT         AND NOT CONFIG_BOOTLOADER_MCUBOOT )
+))
+      message(WARNING "
+        -------------------------------------------------------------------
+        --- WARNING: Using a single bootloader without pm_static.yml.   ---
+        --- Static partitioning requires using a single bootloader when ---
+        --- releasing firmware. Only use dynamic during development.    ---
```

Comment:
I don't get it. Isn't it just as problematic having two bootloaders without a static configuration? You need to statically define the size of the first non-upgradeable bootloader in both cases, to avoid run-time issues when programming the updateable bootloader and/or application.

Reply:
Ah right.
Yes I agree, the non-upgradable parts should be static, also for two bootloaders.
Should I expand this warning to all build with bootloaders then?
(And rephrase it)